src/main/java/com/management/HumanResources/service/UpdateService.java

```diff
@@ -48,35 +50,76 @@ public void eraseAllEmployeeData(long id) {
         firebase.eraseRecord("/time/" + id + ".json");
     }
 
-    public String approveTimeOff(EmployeeTimeOff time) {
+    public String approveTimeOff(int timeOffId) {
+        List<EmployeeTimeOff> employeeTimeOffs = readController.getEmployeeTimeoffs(false);
+        Optional<EmployeeTimeOff> optionalTime = employeeTimeOffs.stream().filter(eto -> eto.getTimeOffId() == timeOffId).findFirst();
+
+        if (!optionalTime.isPresent()) {
+            return "Time off not found";
+        }
+
+        EmployeeTimeOff time = optionalTime.get();
         EmployeeTime employee = parseService.jsonToEmployeeTime(new JSONObject(firebase.getEmployeeTime(time.getEmployeeId())));
+        
         if (employee == null) {
             return "Employee not found";
         }
 
         try {
-            timeOffService.validateTimeOffForEmployee(time, employee);
             employee = timeOffService.approveTimeOffForEmployee(time, employee);
             firebase.updateTimeOff(parseService.timeOffToCsv(employee.getTimeOffs()), employee.getEmployeeId());
             firebase.updateHoursRemaining(employee.getHoursRemaining(), employee.getEmployeeId());
-            return "Time Off approved successfully";
-        } catch (InvalidTimeOffException e) {
+            return "Time off approved successfully";
+        } catch (Exception e) {
             return e.getMessage();
         }
     }
 
-    public String denyTimeOff(EmployeeTimeOff time) {
+    public String denyTimeOff(int timeOffId) {
+        List<EmployeeTimeOff> employeeTimeOffs = readController.getEmployeeTimeoffs(false);
+        Optional<EmployeeTimeOff> optionalTime = employeeTimeOffs.stream().filter(eto -> eto.getTimeOffId() == timeOffId).findFirst();
+
+        if (!optionalTime.isPresent()) {
+            return "Time off not found";
+        }
+
+        EmployeeTimeOff time = optionalTime.get();
         EmployeeTime employee = parseService.jsonToEmployeeTime(new JSONObject(firebase.getEmployeeTime(time.getEmployeeId())));
+        
         if (employee == null) {
             return "Employee not found";
         }
 
         try {
-            timeOffService.validateTimeOffForEmployee(time, employee);
             employee = timeOffService.declineTimeOff(time, employee);
             firebase.updateTimeOff(parseService.timeOffToCsv(employee.getTimeOffs()), employee.getEmployeeId());
-            return "Time Off denied successfully";
-        } catch (InvalidTimeOffException e) {
+            return "Time off denied successfully";
+        } catch (Exception e) {
+            return e.getMessage();
+        }
+    }
+
+    public String removeTimeOff(int timeOffId) {
```

Comment:
Lol so this might be a topic of discussion, this method is functionally identical to `denyTimeOff()` in that they both remove the time off from the employee. All time offs in the database don't have a `isReviewed` attribute. I'm sure we would write some temporary code that goes through each persons time off to add this. But then that might add some complexity to the client such as how would a reviewed time off be displayed? Then some logic has to be written to make sure a reviewed time off isn't denied again or approved 😆. I'll leave it up to you to decide if it's worth it to pursue the reviewed scenario.

Reply:
Yep, you are right. I had some issues when I started the sprint of getting validation errors upon approval/denial of requests but I later completely removed the validation as I thought it was unnecessary. I will remove this remove functionality.

Reply:
Nice, and thank you!